src/LAMMPS.jl

```diff
@@ -1,29 +1,28 @@
 module LAMMPS
 
+using MPI
+
 include("api.jl")
 
 export LMP, command, get_natoms, extract_atom, extract_compute, extract_global,
        gather_atoms
 
+
 mutable struct LMP
-    handle::Ptr{Cvoid}
+    args::Vector{String}
+    comm::Union{MPI.Comm, Nothing}
+    handle::LAMMPSPtr
 
-    function LMP(args::Vector{String}=String[])
-        if isempty(args)
-            argsv = C_NULL
-        else
-            args = copy(args)
-            pushfirst!(args, "lammps")
-            argsv = map(pointer, args)
-        end
+    function LMP(args::Vector{String}=String[], comm = MPI.COMM_WORLD)
+        ptrref = Ref{LAMMPSPtr}(C_NULL)
```

Comment:
I think you still need to `pushfirst!(args, "lammps")`. IIRC lammps expects the binary name as the first arg.